benchmark_runner/workloads/bootstorm_vm.py

```diff
@@ -0,0 +1,123 @@
+            self._verify_elasticsearch_data_uploaded(index=self.__es_index, uuid=self._uuid)
+
+    def __delete_vm_scale(self, vm_num: str):
+        """
+        This method delete vm in parallel
+        """
+        vm_name = f'bootstorm-vm-{self._trunc_uuid}-{vm_num}'
+        self._oc.delete_vm_sync(
+            yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}_{vm_num}.yaml'),
+            vm_name=vm_name)
+
+    @logger_time_stamp
+    def run(self):
+        """
+        This method runs the workload
+        :return:
+        """
+        try:
+            self.__name = self._workload
+            if self._run_type == 'test_ci':
+                self.__es_index = 'bootstorm-test-ci-results'
+            else:
+                self.__es_index = 'bootstorm-results'
+            self.__workload_name = self._workload.replace('_', '-')
+            self.__vm_name = f'{self.__workload_name}-{self._trunc_uuid}'
+            self.__kind = 'vm'
+            self._environment_variables_dict['kind'] = 'vm'
+            if not self._scale:
+                self._oc._create_async(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}.yaml'))
+                self._oc.wait_for_vm_status(vm_name=f'bootstorm-vm-{self._trunc_uuid}', status=VMStatus.Stopped)
+                self.set_bootstorm_vm_start_time(vm_name=self.__vm_name)
+                self._oc.start_vm_sync(vm_name=self.__vm_name)
+                self.__data_dict = self.get_bootstorm_vm_end_time(vm_name=self.__vm_name)
+                self.__status = 'complete' if self.__data_dict else 'failed'
+                if self._es_host:
+                    # upload several run results
+                    self._upload_to_elasticsearch(index=self.__es_index, kind=self.__kind, status=self.__status, result=self.__data_dict)
+                    # verify that data upload to elastic search according to unique uuid
+                    self._verify_elasticsearch_data_uploaded(index=self.__es_index, uuid=self._uuid)
+                self._oc.delete_vm_sync(
+                    yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}.yaml'),
+                    vm_name=self.__vm_name)
+            # scale
+            else:
+                # create namespace
+                self._oc._create_async(yaml=os.path.join(f'{self._run_artifacts_path}', 'namespace.yaml'))
+                # create redis and state signals
+                proc = []
+                scale = int(self._scale)
+                self.__data_dict['scale'] = scale
+                for target in [self.__create_vm_scale, self.__run_vm_scale, self.__delete_vm_scale]:
+                    # create vms in parallel
+                    count = 0
+                    for scale_node in range(len(self._scale_node_list)):
+                        for scale_num in range(scale):
+                            count += 1
+                            p = Process(target=target, args=(str(count),))
```

Comment:
What is the trailing comma in the Process() argument list for?

Reply:
This is the right syntax for passing 1 or more arguments for threading

Reply:
OK.